ios/KlippaScannerSDK.swift

```diff
@@ -0,0 +1,440 @@
+            return
+        }
+
+        guard let modelFile = model["modelFile"] else {
+            return
+        }
+
+        guard let modelLabels = model["modelLabels"] else {
+            return
+        }
+
+        builder.klippaObjectDetectionModel = KlippaObjectDetectionModel(
+            modelFile: modelFile,
+            modelLabels: modelLabels
+        )
+
+    }
+
+    fileprivate func setBuilderCameraModes(_ config: [String: Any], _ builder: KlippaScannerBuilder) {
+        var modes = [KlippaDocumentMode]()
+
+        if let cameraModeSingle = config["CameraModeSingle"] as? [String: Any] {
+
+            let single = KlippaSingleDocumentMode()
+
+            if let name = cameraModeSingle["name"] as? String {
+                single.name = name
+            }
+
+            let instructions = Instructions(message: "Single Document",
+                                            dismissHandler: {
+                self.singleDocumentModeInstructionsDismissed = true
+            })
+
+            if let message = cameraModeSingle["message"] as? String {
+                instructions.message = message
+            }
+
+            modes.append(single)
+        }
+
+        if let cameraModeMulti = config["CameraModeMulti"] as? [String: Any] {
+
+            let multi = KlippaMultipleDocumentMode()
+
+            if let name = cameraModeMulti["name"] as? String {
+                multi.name = name
+            }
+
+            let instructions = Instructions(message: "Multiple Documents",
+                                            dismissHandler: {
+                self.multiDocumentModeInstructionsDismissed = true
+            })
+
+            if let message = cameraModeMulti["message"] as? String {
+                instructions.message = message
+            }
+
+            multi.instructions = instructions
+
+            modes.append(multi)
+        }
+
+        if let cameraModeSegmented = config["CameraModeSegmented"] as? [String: Any] {
+
+            let segmented = KlippaSegmentedDocumentMode()
+
+            if let name = cameraModeSegmented["name"] as? String {
+                segmented.name = name
+            }
+
+            let instructions = Instructions(message: "Segmented Document",
+                                            dismissHandler: {
+                self.segmentedDocumentModeInstructionsDismissed = true
+            })
+
+            if let message = cameraModeSegmented["message"] as? String {
+                instructions.message = message
+            }
+
+            segmented.instructions = instructions
+
+            modes.append(segmented)
+        }
+
+        var index = 0
+        if let startingIndex = config["StartingIndex"] as? Int {
+            index = startingIndex
+        }
+
+        let cameraModes = KlippaCameraModes(
+            modes: modes,
+            startingIndex: index
+        )
+
+        builder.klippaCameraModes = cameraModes
```

Comment:
If modes is empty we should not use this logic and simply have the SDK set the default modes.